### ISSUE TITLE
Timeout issues when calling `/oauth/token` and/or `.well-known/openid-configuration`

### ISSUE THREAD

**Issue by Will956** [auth0/express-openid-connect#482]
### Checklist

- [X] The issue can be reproduced in the [express-openid-connect sample app](https://github.com/auth0-samples/auth0-express-webapp-sample/tree/master/01-Login) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/express-openid-connect#readme), [Examples](https://github.com/auth0/express-openid-connect/blob/master/EXAMPLES.md), and [FAQ](https://github.com/auth0/express-openid-connect/blob/master/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/express-openid-connect/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/express-openid-connect/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Hello 👋 

We currently experiment random timeout issues when login or refreshing the access token on our website. This mainly occurs when the SDK creates the client and fetches our `.well-known/openid-configuration` to then call our `/oauth/token/` endpoint.

The error ⬇️ 
<img width="439" alt="Capture d’écran 2023-08-24 à 17 51 27" src="https://github.com/auth0/express-openid-connect/assets/13664270/180babbf-b289-4d7b-b64e-0fb0e13916fc">

We use Cloudfront as a reverse proxy for our custom domain, and here are the logs when the timeout happens:
```
#Fields: date time x-edge-location sc-bytes c-ip cs-method cs(Host) cs-uri-stem sc-status cs(Referer) cs(User-Agent) cs-uri-query cs(Cookie) x-edge-result-type x-edge-request-id x-host-header cs-protocol cs-bytes time-taken x-forwarded-for ssl-protocol ssl-cipher x-edge-response-result-type cs-protocol-version fle-status fle-encrypted-fields c-port time-to-first-byte x-edge-detailed-result-type sc-content-type sc-content-len sc-range-start sc-range-end
2023-08-24	14:36:44	FRA56-P3	0	[redacted]	POST	[redacted].cloudfront.net	/oauth/token	000	-	express-openid-connect/2.16.0	-	-	Error	[redacted]	auth.[redacted].fr	https	799	4.987	-	TLSv1.3	TLS_AES_128_GCM_SHA256	Error	HTTP/1.1	-	-	28450	0.000	ClientCommError	-	-	-	-
2023-08-24	14:36:44	FRA56-P3	0	[redacted]	GET	[redacted].cloudfront.net	/.well-known/jwks.json	000	-	express-openid-connect/2.16.0	-	-	Error	[redacted]	auth.[redacted].fr	https	500	4.992	-	TLSv1.3	TLS_AES_128_GCM_SHA256	Error	HTTP/1.1	-	-	59060	0.000	ClientCommError	-	-	-	-
```

We can see that Auth0 returns 0 bytes (or the client is unable to get them) and the client closes the connection after 5 seconds (default timeout configuration from that SDK).

Sometimes we also have another error: `Issuer.discover() failed.` that is linked to the timeout error (but not all the time).

For the `Issuer.discover() failed.`, we are aware that `agent-base` can be an issue. We checked our version and that looks fine:
```
$ npm list agent-base
project@1.0.0 /project
└─┬ socks-proxy-agent@8.0.1
  └── agent-base@7.1.0
```

That's quite odd and we can't properly reproduce it.

I'm wondering if that is something that is on our side or on the Auth0 side? Do you have any clues? Thanks a lot 🙏 

### Reproduction

Can't reproduce it, it's fully random 😢 

### express-openid-connect version

2.16.0

### Express version

4.18.2

### Node.js version

18.12.1

**Comment by adamjmcgrath** [comment]
Hi @Will956 

> Sometimes we also have another error: Issuer.discover() failed. that is linked to the timeout error (but not all the time).

I would expect you to get the `Issuer.discover() failed.` error every time the request to `.well-known/openid-configuration` fails.

> For the Issuer.discover() failed., we are aware that agent-base can be an issue. 

If you're referring to the issue I'm thinking of, this would fail all the time - not intermittently.

I'm not sure how intermittent the random timeouts are, but it looks like some of your requests are taking longer than 5 secs and timing out - have you tried increasing the `httpTimeout` setting to something longer, like 10 secs and seeing if this reduces the frequency of timeouts?

Ultimately, if the Authorization server takes > 5 secs to respond and your timeout settings are 5 secs, you will get an error (the error will be different depending on what you're trying to do; exchange the code, read the AS's discovery document, etc.)

**Comment by Will956** [comment]
Hey 👋 

Thanks for the quick answer, 

> I would expect you to get the Issuer.discover() failed. error every time the request to .well-known/openid-configuration fails.

Yes you are right, this may be hidden by our error tracker provider for some reason but this error is present every time.

> If you're referring to the issue I'm thinking of, this would fail all the time - not intermittently.

Ok, so that's not the problem here thanks for the clarification.

> have you tried increasing the httpTimeout setting to something longer, like 10 secs and seeing if this reduces the frequency of timeouts?

Yes we're going to increase that value to better investigate that issue, will keep up updated on that.

The authorization server is Auth0, so maybe we should open a ticket directly with the support to investigate why. Here are additional logs:
<img width="887" alt="Capture d’écran 2023-08-28 à 10 24 39" src="https://github.com/auth0/express-openid-connect/assets/13664270/6d2c7c77-d954-4b66-b4ac-fb5197a80932">

As we can see, the time taken is quite big for some reasons and we also noticed that CloudFront has no cache every time.

We followed this guide https://auth0.com/docs/customize/custom-domains/self-managed-certificates/configure-aws-cloudfront-for-use-as-reverse-proxy to set up caching in CloudFront, but as we can see, the cache is always missed 🤔 I would expect CloudFront to have a caching version of our `.well-known/openid-configuration` at least, no ?

Thanks 🙏
